src/cryptography/x509/extensions.py

```diff
@@ -102,6 +103,14 @@ def __iter__(self):
     def __len__(self):
         return len(self._extensions)
 
+    def __getitem__(self, idx):
+        if isinstance(idx, slice):
+            start, stop, step = idx.indices(len(self))
+            return self._extensions[start:stop:step]
+        else:
+            idx = operator.index(idx)
+            return self._extensions[idx]
```

Comment:
I think for this class just `return self._extensions[idx]` is all you need, you can drop all the other logic.